deepchem/models/losses.py

```diff
@@ -205,6 +205,114 @@ def loss(output, labels):
     return loss
 
 
+class VAE_ELBO(Loss):
+  """The Variational AutoEncoder loss, KL Divergence Regularize + marginal log-likelihood.
+  
+  The logvar and mu should have shape (batch_size, hidden_space).
+  The x and reconstruction_x should have (batch_size, attribute).
+  kl_scale: KLD regularized weights.
+  """
+
+  def _compute_tf_loss(self, logvar, mu, x, reconstruction_x, kl_scale = 1):
+    import tensorflow as tf
+    x, reconstruction_x = _make_tf_shapes_consistent(x, reconstruction_x)
+    x, reconstruction_x = _ensure_float(x, reconstruction_x)
+    BCE = tf.keras.losses.binary_crossentropy(x, reconstruction_x)
+    KLD = VAE_KLDivergence()._compute_tf_loss(logvar, mu)
+    return BCE + kl_scale*KLD
+
+  def _create_pytorch_loss(self):
+    import torch
+    bce = torch.nn.BCELoss(reduction='none')
+
+    def loss(logvar, mu, x, reconstruction_x, kl_scale = 1):
+      x, reconstruction_x = _make_pytorch_shapes_consistent(x, reconstruction_x)
+      BCE = torch.mean(bce(x, reconstruction_x), dim=-1)
+      KLD = (VAE_KLDivergence()._create_pytorch_loss())(logvar, mu)
+      return BCE + kl_scale*KLD
+
+    return loss
+
+
+class VAE_KLDivergence(Loss):
+  """The KL_divergence between hidden distribution and normal distribution
+  The logvar should have shape (batch_size, hidden_space) and each term represents
+  standard deviation of hidden distribution. The mean shuold have 
+  (batch_size, hidden_space) and each term represents mean of hidden distribtuon.
```

Comment:
Could you add references/examples to this docstring as indicated in the comments above?